### ISSUE TITLE
Confusing subscriber management from email

### ISSUE THREAD

**Issue by simison** [Automattic/wp-calypso#81025]
### Quick summary

Subscriber management shows confusing results when clicking it open from an email.

### Steps to reproduce

1. Subscribe to a blog, receive email:
    <img width="835" alt="image" src="https://github.com/Automattic/wp-calypso/assets/87168/66fe291d-9bc9-48ea-8ed3-d1d18bbf4e45">
2. Follow the link, end up at https://wordpress.com/read/subscriptions?s=developer.wordpress.com
3. See following screens some seconds apart after each:
    
<img width="1351" alt="Screenshot 2023-08-24 at 15 50 42" src="https://github.com/Automattic/wp-calypso/assets/87168/164da155-feb2-46be-87c3-5ed414f8c4a0">
<img width="1338" alt="Screenshot 2023-08-24 at 15 50 44" src="https://github.com/Automattic/wp-calypso/assets/87168/e081e821-9963-4fcb-99ed-a00f43f146ff">
<img width="1349" alt="Screenshot 2023-08-24 at 15 50 46" src="https://github.com/Automattic/wp-calypso/assets/87168/5bdbb1ae-5ea6-4663-bbe7-e0afa014177e">
<img width="1349" alt="Screenshot 2023-08-24 at 15 50 57" src="https://github.com/Automattic/wp-calypso/assets/87168/1c0100be-01c6-474a-b9c5-567fa2217a1c">

- Very buggy with search "no results", and then a result which isn't actually what I can manage like email promised, and then finally the management thing.
- Seeing duplicate is confusing. I didn't want to search anything, I just wanted to manage this one subscription.
- We have the individual site's management screen. Why not send me there directly?
- Spinner loader; grey "placeholder" loader in the shape of potential content feel better, feel faster, and help predict the content that might appear there.

### What you expected to happen

—

### What actually happened

—

### Impact

All

### Available workarounds?

No but the platform is still usable

**Comment by mashikag** [comment]
> - Seeing duplicate is confusing. I didn't want to search anything, I just wanted to manage this one subscription.

Unfortunately this is due to elastic-search not being in sync with the real data in time. You must have opened this screen quite soon after subscribing.  Usually it takes few minutes for elastic-search to sync up. At the time of the development, it seemed like a quick "solution" to the problem. Especially that this is the way it showed up in the old screen (https://wordpress.com/following/manage).

Never-the-less, it is not a good UX, and it should be fixed.

**Comment by simison** [comment]
> You must have opened this screen quite soon after subscribing.

That's that I'd expect people do when they get email with link to this screen right after subscribing. :-)

Couldn't we just link to individual subscription page (p7DVsv-huC-p2)?

And recognice in Calypso code that we're showing the same site twice — and just not show it twice then?

**Comment by cuemarie** [comment]
📌 REPRODUCTION RESULTS
- Tested on Simple – Replicated somewhat

📌 FINDINGS/SCREENSHOTS/VIDEO
- Replicated the 2 sites appearing in search results in my Reader; I did not run into the first issue of the site not appearing at all - seems I wasn't quite quick enough!
- I subscribed to developer.wordpress.com, and probably 30 seconds later clicked on the `Manage Subscriptions` email, and this is what loaded:

https://github.com/Automattic/wp-calypso/assets/27249804/8a50af19-f317-4a18-9328-778bf12d22eb

📌 ACTIONS
- Set as `To Do`
- Dropped priority to Normal
- @Automattic/apex

**Comment by simison** [comment]
I also noticed here we loaded the site icon only once for the same site:

<img width="1380" alt="Screenshot 2023-08-24 at 20 58 26" src="https://github.com/Automattic/wp-calypso/assets/87168/77fac80c-dea4-43be-9cdd-afc2f34c3acc">

**Comment by mashikag** [comment]
> Couldn't we just link to individual subscription page (p7DVsv-huC-p2)?

We thought about doing so during the development process. However, we could not afford increasing the scope of the project (pdDOJh-1S1-p2). The subscription details screen was initially implemented with the external subscriptions management portal (https://wordpress.com/subscriptions) in mind. It would require some additional work to integrate it into reader. Instead, we decided to replicate the behavior that existed with the old "Manage followed" page, i.e. to display the "Manage subscribed sites" page with the domain name of the site provided as search param in the URL.

cc @yansern

**Comment by mashikag** [comment]
> Spinner loader; grey "placeholder" loader in the shape of potential content feel better, feel faster, and help predict the content that might appear there.

Related conversation: p1689166571025639-slack-C02TCEHP3HA

**Comment by mashikag** [comment]
> Couldn't we just (...)
> (...) recognice in Calypso code that we're showing the same site twice — and just not show it twice then?

We can, and this is kind of what was done, but instead of hiding the search result we display it with disabled "Subscribed" button once we discover that the site belongs to subscribed sites. This decision was partially driven by the fact that the displaying of the subscribed-to sites vs. search results (which, in theory, should not contain subscribed-to sites) depend on two different REST API requests. One may return, before the other one does without any guarantees. However, I cannot re-call the rest of the technical factors that influenced this UI decision, so I will revisit the code and will see what can be done to hide the duplicate result, as per your suggestion.

**Comment by yansern** [comment]
> > Couldn't we just link to individual subscription page (p7DVsv-huC-p2)?
> 
> We thought about doing so during the development process. However, we could not afford increasing the scope of the project (pdDOJh-1S1-p2). The subscription details screen was initially implemented with the external subscriptions management portal (https://wordpress.com/subscriptions) in mind. It would require some additional work to integrate it into reader. Instead, we decided to replicate the behavior that existed with the old "Manage followed" page, i.e. to display the "Manage subscribed sites" page with the domain name of the site provided as search param in the URL.
> 
> cc @yansern

My intention is to tie this in as part of (or as side work to) the "Newsletter Categories: Subscription & Subscriber Management Integration" sub-project. Context: pdDOJh-2gB-p2

**Comment by mashikag** [comment]
> I also noticed here we loaded the site icon only once for the same site:
> 
> <img alt="Screenshot 2023-08-24 at 20 58 26" width="1380" src="https://user-images.githubusercontent.com/87168/263063373-77fac80c-dea4-43be-9cdd-afc2f34c3acc.png">

This is another issue that we were aware of before we moved on to our next priorities. 🙊  The issue is related to the fact that different endpoints are used to fetch the data for the two lists. Unfortunately they not always return the same results when it comes to the site icons. 

- The list of a user's current site subscriptions is fetched via a call to: `https://public-api.wordpress.com/rest/v1.2/read/following/mine`
- The search results data is first returned by a call to: `https://public-api.wordpress.com/rest/v1.1/read/feed`. 
   - The result of the above call, does not return all the data required for displaying a site row. Therefore, for every site result we got, we do a follow up request to: `https://public-api.wordpress.com/rest/v1.1/read/sites/$siteId`

Often times some sites returned by a call to `https://public-api.wordpress.com/rest/v1.2/read/following/mine` are returned with no site icon, suggesting they do not have one. However, a call to `https://public-api.wordpress.com/rest/v1.1/read/sites/$siteId` for the same site reveals that the site actually has an icon. Therefore, the issue would ideally be fixed by addressing the site icon retrieval done by the endpoint: `https://public-api.wordpress.com/rest/v1.2/read/following/mine`. 

_Note: As part of the project, we did not develop, nor make any changes to the `https://public-api.wordpress.com/rest/v1.2/read/following/mine` endpoint. We just re-used it the way it used to be._

**Comment by mashikag** [comment]
> > Spinner loader; grey "placeholder" loader in the shape of potential content feel better, feel faster, and help predict the content that might appear there.
> 
> Related conversation: p1689166571025639-slack-C02TCEHP3HA

I spent some time looking into this today. It is a bit trickier than it may initially seem. 

On our subscription management page, we leverage react-query for local caching. This setup ensures that when a user revisits the page, we can swiftly display the subscription list from the cache. However, even with cached data, a fresh network request is initiated in the background. If the new response differs from our cached data, we re-render the list.

Given this behavior, it's ideal to implement two loading indications:
1. **Site Subscription Row Placeholder(s)**
   - Intended for first-time visitors who don't yet have any cache data.
2. **Subtle Spinner**
    - Aimed at returning users with populated cache data.
    - While the cached data is displayed immediately, this spinner informs the user that a refresh is in progress, and the content may update upon completion of the data fetch.

It's crucial to understand that we display two distinct lists on the screen:

1. **Search Results for User's Subscriptions**
2. **Search Results for Other Sites (those not subscribed to)**

![BTD1FC.png](https://github.com/Automattic/wp-calypso/assets/2019970/ea69fc66-f186-41b6-8df2-87c968971574)

Each list is populated by a separate query. As a result, one list might load faster than the other, and there's no guaranteed synchronization between them. The loading solution previously discussed is effective for the first list but poses challenges for the second.

Firstly, the list displaying results for sites the user hasn't subscribed to lacks an "empty state." This means that if a search yields no matches for other sites, we currently don't notify the user that we've attempted to fetch results. Instead, we leave the list's space blank. 

To illustrate this, consider the screenshot below. The first image shows a search term that doesn't match any of my current subscriptions but returns results from other sites. The second image reveals a search that matches no sites, whether subscribed to or not.

![cBsmjC.png](https://github.com/Automattic/wp-calypso/assets/2019970/1c12d914-cdb7-47d6-a9d0-8e654fb116b5)

Given the absence of an empty state for the "other sites" list, we need to decide if it's appropriate to show a loading indicator for this list.

@simison, @yansern, I'd appreciate your input on this. What are your thoughts?

cc @saygunnyc, @davemart-in

**Comment by mashikag** [comment]
Another issue that I noticed when I tried to repro the issue, is that the only time I was able to receive the "Welcome to Developer Resources!" email, was when I subscribed using the "Follow" button. 

![dNNd9y.png](https://github.com/Automattic/wp-calypso/assets/2019970/dd18cd13-1d03-4334-b012-c1c0ca957824)

____
When I subscribed in Reader, I received no email notification/confirmation what so ever.

![7LeEqU.png](https://github.com/Automattic/wp-calypso/assets/2019970/3738ae00-1cd7-4c50-91f2-15110095ba64)

____
When I subscribed (to a different site) using our Subscribe block, instead I received "Confirm your subscription for %s" email.
![Ba0Jgk.png](https://github.com/Automattic/wp-calypso/assets/2019970/7063e697-77d4-4753-90ea-0cb89905b73b)
____
Summarizing, three different ways to subscribe result in different emails being received by the subscriber.

**Comment by mashikag** [comment]
Another issue that I noticed when I tried to repro the issue, is that the only time I was able to receive the "Welcome to Developer Resources!" email, was when I subscribed using the "Follow" button. 

![dNNd9y.png](https://github.com/Automattic/wp-calypso/assets/2019970/dd18cd13-1d03-4334-b012-c1c0ca957824)

____
When I subscribed in Reader, I received no email notification/confirmation what so ever.

![7LeEqU.png](https://github.com/Automattic/wp-calypso/assets/2019970/3738ae00-1cd7-4c50-91f2-15110095ba64)

____
When I subscribed (to a different site) using our Subscribe block, instead I received "Confirm your subscription for %s" email.
![Lqm2d4.png](https://github.com/Automattic/wp-calypso/assets/2019970/e458915e-b997-4692-a94d-8cd485a61da9)
____
Summarizing, three different ways to subscribe result in different emails being received by the subscriber.

**Comment by mashikag** [comment]
> > I also noticed here we loaded the site icon only once for the same site:
> > <img alt="Screenshot 2023-08-24 at 20 58 26" width="1380" src="https://user-images.githubusercontent.com/87168/263063373-77fac80c-dea4-43be-9cdd-afc2f34c3acc.png">
> 
> This is another issue that we were aware of before we moved on to our next priorities. 🙊 The issue is related to the fact that different endpoints are used to fetch the data for the two lists. Unfortunately they not always return the same results when it comes to the site icons.
> 
> * The list of a user's current site subscriptions is fetched via a call to: `https://public-api.wordpress.com/rest/v1.2/read/following/mine`
> * The search results data is first returned by a call to: `https://public-api.wordpress.com/rest/v1.1/read/feed`.
>   
>   * The result of the above call, does not return all the data required for displaying a site row. Therefore, for every site result we got, we do a follow up request to: `https://public-api.wordpress.com/rest/v1.1/read/sites/$siteId`
> 
> Often times some sites returned by a call to `https://public-api.wordpress.com/rest/v1.2/read/following/mine` are returned with no site icon, suggesting they do not have one. However, a call to `https://public-api.wordpress.com/rest/v1.1/read/sites/$siteId` for the same site reveals that the site actually has an icon. Therefore, the issue would ideally be fixed by addressing the site icon retrieval done by the endpoint: `https://public-api.wordpress.com/rest/v1.2/read/following/mine`.
> 
> _Note: As part of the project, we did not develop, nor make any changes to the `https://public-api.wordpress.com/rest/v1.2/read/following/mine` endpoint. We just re-used it the way it used to be._

This is being followed up in: D119983-code

**Comment by davemart-in** [comment]
I don't think this belongs in The One Board. This is too big of an issue to realistically expect someone to pick up on a groundskeeper rotation.

Apex has moved on from working on this screen I believe.

@simison can we move this to Zap's team board to investigate?
